app/api/domains/api.py

```diff
@@ -108,6 +109,35 @@ def format_map_basic(m: Beatmap) -> dict[str, object]:
     }
 
 
+@router.get("/search_players")
+async def api_search_players(
+    search: Optional[str] = Query(None, alias="q", min=2, max=32),
+    db_conn: databases.core.Connection = Depends(acquire_db_conn),
+):
+    """Search for users on the server by name."""
+    rows = await db_conn.fetch_all(
+        "SELECT id, name FROM users WHERE name LIKE :search AND priv & 3 = 3 ORDER BY id ASC",
+        {"search": f"%{search}%" if search is not None else None},
+    )
+
+    if len(rows) == 0:
+        return ORJSONResponse(
+            {
+                "status": "error",
+                "message": "nothing was found matching those parameters!",
+            },
+            status_code=status.HTTP_404_NOT_FOUND,
+        )
+    else:
+        return ORJSONResponse(
+            {
+                "status": "success",
+                "results": len(rows),
+                "result": [dict(row) for row in rows],
+            },
+        )
```

Comment:
```suggestion
    return ORJSONResponse(
        {
            "status": "success",
            "results": len(rows),
            "result": [dict(row) for row in rows],
        },
    )
```

Reply:
why always return 200?

Reply:
no point in doing more serverside work

Reply:
this is more standard in rest

when fetching an array of resources, an empty array is still a valid response
an example might be, e.g. `GET /users` returning `200 OK` with body `[]`

404 is more for the case where you're looking up a specific resource and it is not found
an example might be, e.g. `GET /users/1` returning `404 NOT FOUND`